plugin/trino-mysql/src/test/java/io/trino/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -625,72 +614,68 @@ public void testTimestampFromMySql(ZoneId sessionZone)
-                .addRoundTrip("timestamp(6)", "TIMESTAMP '1970-01-01 00:00:01.12345649999'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
-                .addRoundTrip("timestamp(6)", "TIMESTAMP '1970-01-01 00:00:01.1234565'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
-                .addRoundTrip("timestamp(6)", "TIMESTAMP '1970-01-01 00:00:01.1234569'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
+                .addRoundTrip("timestamp(0)", "TIMESTAMP '1970-01-01 01:00:01'", createTimestampType(0), "TIMESTAMP '1970-01-01 01:00:01'")
+                .addRoundTrip("timestamp(1)", "TIMESTAMP '1970-01-01 01:00:01.1'", createTimestampType(1), "TIMESTAMP '1970-01-01 01:00:01.1'")
+                .addRoundTrip("timestamp(2)", "TIMESTAMP '1970-01-01 01:00:01.12'", createTimestampType(2), "TIMESTAMP '1970-01-01 01:00:01.12'")
+                .addRoundTrip("timestamp(3)", "TIMESTAMP '1970-01-01 01:00:01.123'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.123'")
+                .addRoundTrip("timestamp(4)", "TIMESTAMP '1970-01-01 01:00:01.1234'", createTimestampType(4), "TIMESTAMP '1970-01-01 01:00:01.1234'")
+                .addRoundTrip("timestamp(5)", "TIMESTAMP '1970-01-01 01:00:01.12345'", createTimestampType(5), "TIMESTAMP '1970-01-01 01:00:01.12345'")
+                .addRoundTrip("timestamp(6)", "TIMESTAMP '1970-01-01 01:00:01.123456'", TIMESTAMP_MICROS, "TIMESTAMP '1970-01-01 01:00:01.123456'")
+                .addRoundTrip("timestamp(6)", "TIMESTAMP '1970-01-01 01:00:01.1234560'", TIMESTAMP_MICROS, "TIMESTAMP '1970-01-01 01:00:01.123456'")
+                .addRoundTrip("timestamp(6)", "TIMESTAMP '1970-01-01 01:00:01.12345649999'", TIMESTAMP_MICROS, "TIMESTAMP '1970-01-01 01:00:01.123456'")
+                .addRoundTrip("timestamp(6)", "TIMESTAMP '1970-01-01 01:00:01.1234565'", TIMESTAMP_MICROS, "TIMESTAMP '1970-01-01 01:00:01.123457'")
+                .addRoundTrip("timestamp(6)", "TIMESTAMP '1970-01-01 01:00:01.1234569'", TIMESTAMP_MICROS, "TIMESTAMP '1970-01-01 01:00:01.123457'")
 
                 // TODO support higher precision timestamps (https://github.com/trinodb/trino/issues/6910)
                 // before epoch with second fraction
                 //.addRoundTrip("timestamp(7)", "TIMESTAMP '1969-12-31 23:59:59.1230000'", createTimestampType(7), "TIMESTAMP '1969-12-31 23:59:59.1230000'")
                 //.addRoundTrip("timestamp(7)", "TIMESTAMP '1969-12-31 23:59:59.1234567'", createTimestampType(7), "TIMESTAMP '1969-12-31 23:59:59.1234567'")
 
                 // precision 0 ends up as precision 0
-                .addRoundTrip("timestamp(0)", "TIMESTAMP '1970-01-01 00:00:01'", TIMESTAMP_MILLIS, "TIMESTAMP '1970-01-01 01:00:01.000'")
+                .addRoundTrip("timestamp(0)", "TIMESTAMP '1970-01-01 01:00:01'", createTimestampType(0), "TIMESTAMP '1970-01-01 01:00:01'")
```

Comment:
Please uncomment the TODO above (for https://github.com/trinodb/trino/issues/6910 issue)